src/main/java/org/folio/services/finance/transaction/PendingPaymentWorkflowService.java

```diff
@@ -160,4 +174,29 @@ protected Transaction buildBaseTransaction(InvoiceWorkflowDataHolder holder) {
             .withExpenseClassId(holder.getExpenseClassId());
   }
 
+  private Future<Void> unreleaseReleasedEncumbrances(List<InvoiceWorkflowDataHolder> holders, RequestContext requestContext) {
+    List<Transaction> previousUnreleasedEncumbrances = holders.stream()
+      .map(InvoiceWorkflowDataHolder::getEncumbrance)
+      .filter(t -> Objects.nonNull(t) && t.getEncumbrance().getStatus() == UNRELEASED)
+      .toList();
+    if (previousUnreleasedEncumbrances.isEmpty()) {
+      log.info("unreleaseReleasedEncumbrances:: no encumbrance to unrelease");
+      return succeededFuture();
+    }
+    List<String> ids = previousUnreleasedEncumbrances.stream().map(Transaction::getId).toList();
+    return baseTransactionService.getTransactionsByIds(ids, requestContext)
+      .compose(newEncumbrances -> {
+        List<Transaction> encumbrancesToUnrelease = newEncumbrances.stream()
+          .filter(t -> t.getEncumbrance().getStatus() == RELEASED)
+          .toList();
+        if (encumbrancesToUnrelease.isEmpty()) {
+          log.info("unreleaseReleasedEncumbrances:: no encumbrance to unrelease");
```

Comment:
this log can have a bit different text from the same on 183 line. In this case we can easily distinguish which if statement that log relates to.
Also some comment can be added to specify that we use that holder that stores encumbrances without an action applied yet that can speedup understanding of this method.

Reply:
OK